### ISSUE TITLE
[Proof] Relay signature & Merkle proof validation

### ISSUE THREAD

**Issue by bryanchriswhite** [pokt-network/poktroll#401]
## Objective

The proof module MUST validate Merkle proofs (i.e. [`MsgSubmitProof#closest_merkle_proof`](https://github.com/pokt-network/poktroll/blob/main/proto/poktroll/proof/proof.proto#L15)) in its [`msgServer#SubmitProof()` method](https://github.com/pokt-network/poktroll/blob/main/x/proof/keeper/msg_server_submit_proof.go#L12).

- May depend on #377.
- **Integration test implementation depends on #404.**

## Origin Document

![image](https://github.com/pokt-network/poktroll/assets/600733/298e9595-5837-4edd-b4bf-85936ff17773)

## Goals

- Validate that the merkle proof's root, path, value hash match on-chain expectations.
- Validate that the relay (proof value) request and response signatures match the application and supplier (respectively) for the given session.

## Deliverables

- [ ] A PR which implements merkle proof validation.
- [ ] A PR which implements relay unpacking and validation.
- [ ] A PR which adds `relay_min_difficulty` to the proof module.
- [ ] A PR which implements relay minimum difficulty validation.

## Non-goals / Non-deliverables

- Refactoring outside the scope of the goals outlined in this issue.
- Additional validation (e.g. distribution).

## General deliverables

- [ ] **Comments**: Add/update TODOs and comments alongside the source code so it is easier to follow.
- [ ] **Testing**: Add new tests (unit and/or E2E) to the test suite.
- [ ] **Makefile**: Add new targets to the Makefile to make the new functionality easier to use.
- [ ] **Documentation**: Update architectural or development READMEs; use [mermaid](https://mermaid-js.github.io/mermaid/) diagrams where appropriate.

---

**Creator**: @bryanchriswhite 
**Co-Owners**:

**Comment by red-0ne** [comment]
Closed as per #406
